content/blog/httr2-1-0-0/index.Rmd

```diff
@@ -0,0 +1,268 @@
+resp |> resp_header("Content-Length")
+
+resp |> resp_headers()
+```
+
+## Error handling
+
+You can use `resp_status()` to see the returned status:
+
+```{r}
+resp |> resp_status()
+```
+
+However, this will almost always be 200, because httr2 automatically follows redirects (values in the 300s) and turns HTTP errors (values in the 400s and 500s) into errors.
+The following example shows what this looks like using a example endpoint that throws the returns the request status:
+
+```{r, error = TRUE}
+req |> 
+  req_url_path("/status/404") |> 
+  req_perform()
+
+req |> 
+  req_url_path("/status/500") |> 
+  req_perform()
+```
+
+httr2 provides two main tools to customise this behaviour:
+
+-   `req_error()` gives you full control over what responses will be considered to be errors, and allows you to put additional information in the body of the error.
+-   `req_retry()` helps deal with transient errors, where you need to wait a bit and try again. For example, many APIs are rate limited and will return a 429 status if you have made too many requests.
+
+You can learn more about both of these functions in `vignette("wrapping-apis")` as they are particularly important when creating an R package (or script) that wraps a web API.
+
+## Control the request process
+
+There are a number of other `req_` functions don't directly affect the HTTP request but instead control the overall process of submitting a request and handling the response.
+These include:
+
+-   `req_cache()` sets up a cache so if repeated requests return the same results, you can avoid a trip to the server.
+    `req_cache()` automatically prunes the cache, ensuring that by default it stays under 1 GB
+
+-   `req_throttle()` automatically adds a small delay before each request so you can avoid hammering a server with many requests.
+
+-   `req_progress()` adds a progress bar for long downloads or uploads.
+
+-   `req_cookie_preserve()` lets you preserve cookies across requests.
+
+Additionally, httr2 provides many helpers for authenticating with OAuth, wrapping many more styles than httr.
+You've probably used OAuth a bunch without knowing what it's called: you use it when you login to a non-Google website using your Google account, when you give your phone access to your twitter account, or when you login to a streaming app on your smart TV.
+OAuth is a big, complex topic, and is documented in `vignette("oauth2")`
+
+## Multiple requests
+
+httr2 includes three functions to perform multiple requests:
+
+-   `req_perform_sequential()` takes a list of requests and performs them in order.
+
+-   `req_perform_parallel()` takes a list of requests and performs them in parallel (up to 6 at a time by default).
+    It's similar to `req_perform_sequential()`, but is obviously faster, at the expense of potentially hammering a server.
+    However, it also has some limitations: most importantly it can't re-request an expired OAuth token and it doesn't respect `req_retry()` or `req_throttle()`.
```

Comment:
```suggestion
    However, it also has some limitations: most importantly it can't refresh an expired OAuth token and it doesn't respect `req_retry()` or `req_throttle()`.
```

Not sure about this. Are you trying to distinguish the case of "valid refresh token, need new access token" from "invalid refresh token, need to do full dance"?